source/developers-guide/backend-statistics-extension/index.md

```diff
@@ -0,0 +1,315 @@
+In the callback method, add your plugin template directory.
+Load your custom ExtJs files by using the different actions.
+Have a look at [this article](/developers-guide/backend-extension/#extending), to learn more about that.
+
+### Backend controller
+
+SwagCustomStatistics/Controllers/Backend/SwagCustomStatistics.php
+
+```php
+<?php
+
+class Shopware_Controllers_Backend_SwagCustomStatistics extends Shopware_Controllers_Backend_ExtJs
+{
+    public function getVoucherStatisticsAction()
+    {
+        $additionalSelects = '';
+        $idList = (string) $this->Request()->getParam('selectedShops');
+
+        if (!empty($idList)) {
+            $selectedShopIds = explode(',', $idList);
+
+            foreach ($selectedShopIds as $shopId) {
+                $additionalSelects .= ', SUM(IF(vouchers.subshopID = ' . $shopId . ', codes.cashed, 0)) as amount' . $shopId . " \n";
+            }
+        }
+
+        $select = '
+            SELECT
+                COUNT(codes.cashed) as amount,
+                vouchers.description as name
+                ' . $additionalSelects . '
+            FROM s_emarketing_voucher_codes codes
+            INNER JOIN s_emarketing_vouchers vouchers
+                ON vouchers.id = codes.voucherID
+            WHERE codes.cashed = 1
+            GROUP BY vouchers.id';
+
+
+        $data = $this->container->get('dbal_connection')->fetchAll($select);
+
+        $this->View()->assign([
+            'success' => true,
+            'data' => $data,
+            'count' => count($data)
+        ]);
+    }
+}
+```
+
+Select the used individuell vouchers in the `getVoucherStatisticsAction`.
+As it is possible to select different shops in the analytics module, consider the selected shops from the request. They are separated by commas.
+The action is called by the [ExtJs](#custom-statistics-store) store to gather the data.
+
+## ExJs implementation
+
+### Custom statistics store
+
+SwagCustomStatistics/Resources/views/backend/analytics/swag_custom_statistics/store/navigation/voucher.js
+
+```javascript
+Ext.define('Shopware.apps.Analytics.swagCustomStatistics.store.navigation.Voucher', {
+    extend: 'Ext.data.Store',
+    alias: 'widget.analytics-store-voucher',
+    remoteSort: true,
+
+    fields: [
+        'amount',
+        'name'
+    ],
+
+    proxy: {
+        type: 'ajax',
+
+        url: '{url controller=SwagCustomStatistics action=getVoucherStatistics}',
+
+        reader: {
+            type: 'json',
+            root: 'data',
+            totalProperty: 'total'
+        }
+    },
+
+    constructor: function(config) {
+        var me = this;
+        config.fields = me.fields;
+
+        if (config.shopStore) {
+            config.shopStore.each(function(shop) {
+                config.fields.push('amount' + shop.data.id);
+            });
+        }
+
+        me.callParent(arguments);
+    }
+});
+```
+
+Define the fields in this store, you want to get access to later in the grid/chart.
```

Comment:
maybe: In this store you can define the fields, which you want to access later in the grid/chart.

Reply:
done